src/OpenLoco/Things/ThingManager.h

```diff
@@ -43,53 +43,54 @@ namespace OpenLoco::ThingManager
     void moveSpriteToList(thing_base* const thing, const thing_list list);
     bool checkNumFreeThings(const size_t numNewThings);
 
-    class VehicleHeadIterator
+    template<typename ThingType, thing_id_t thing_base::*nextList>
```

Comment:
The only thing bothering me is that we pass the offset of next_thing_id to this template. Is there no way to specify that `ThingType` needs to be a child of `thing_base`?

Reply:
hmm could do it with SIFINAE or contracts but do we really want to do that.

Reply:
If it overcomplicates it, don't bother. Other stuff looks alright